### PR TITLE
Reader: reduxify streams

### PR DESCRIPTION
**Pull request by samouri**
this is some basic redux actions/reducer/data-layer. 
It is a starting point for reduxifying streams.

Currently it is unhooked and will not be shipped. 
No edge-cases are being handled

To Test
------
- unit tests should pass, but circle ci will tell you that.
- architecture should make sense in that you can envision this as a building block towards getting fully reduxified streams integrated

### CODE REVIEW

File: client/state/reader/streams/schema.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/** @format */
+
+export const itemsSchema = {
+	type: 'object',
+	patternProperties: {
+		'(w|d)+': {
```
Comment (blowery) [inline client/state/reader/streams/schema.js]: is that a valid pattern?

Comment (samouri) [reply client/state/reader/streams/schema.js]: it worked in my (too limited) test cases! 
After checking on https://www.jsonschemavalidator.net/ I think a better pattern would be:
`\\w|\\d`. This would match anything containing either letters or numbers.
